internal/internal_decision_state_machine.go

```diff
@@ -618,13 +618,15 @@ func (d *signalExternalWorkflowDecisionStateMachine) handleCompletionEvent() {
 	}
 }
 
-func (d *markerDecisionStateMachine) handleCompletionEvent() {
-	// Marker decision transit from SENT to COMPLETED on EventType_MarkerRecorded event
+func (d *markerDecisionStateMachine) handleDecisionSent() {
```

Comment:
Wouldn't introducing another type of marker state machine for this use case make it more clear?

Reply:
All 3 maker use cases (SideEffect/Version/LocalActivity) are all the same, and this change serves them well.  Would add new type when needed in future.